pkg/predicate/predicate.go

```diff
@@ -49,3 +52,40 @@ func (OnlyUpdateWhenGenerationNotChanged) Delete(e event.DeleteEvent) bool {
 func (OnlyUpdateWhenGenerationNotChanged) Generic(e event.GenericEvent) bool {
 	return false
 }
+
+// LabelsAndGenerationPredicate is based on the default predicate functions but overrides the Update function
+// to only return true if either the labels or generation have changed, status changes won't cause reconciliation
+type LabelsAndGenerationPredicate struct {
+	predicate.Funcs
+}
+
+// Update only returns true if either the labels or generation have changed
+func (LabelsAndGenerationPredicate) Update(e event.UpdateEvent) bool {
+
+	if e.MetaOld == nil {
+		log.Error(nil, "Update event has no old metadata", "event", e)
+		return false
+	}
+	if e.ObjectOld == nil {
+		log.Error(nil, "Update event has no old runtime object to update", "event", e)
+		return false
+	}
+	if e.ObjectNew == nil {
+		log.Error(nil, "Update event has no new runtime object for update", "event", e)
+		return false
+	}
+	if e.MetaNew == nil {
+		log.Error(nil, "Update event has no new metadata", "event", e)
+		return false
+	}
+
+	// reconcile if the labels have changed
+	if !reflect.DeepEqual(e.MetaOld.GetLabels(), e.MetaNew.GetLabels()) {
+		return true
+	}
+
+	if e.MetaNew.GetGeneration() == e.MetaOld.GetGeneration() && e.MetaNew.GetGeneration() != 0 {
+		return false
+	}
+	return true
```

Comment:
or, in case you dropped the `e.MetaNew.GetGeneration() != 0` then it could be simplified to:
```suggestion
	return !reflect.DeepEqual(e.MetaOld.GetLabels(), e.MetaNew.GetLabels()) ||
		e.MetaNew.GetGeneration() != e.MetaOld.GetGeneration()
```

Reply:
It was based on the one from operator sdk: https://github.com/operator-framework/operator-sdk/blob/ef0bc885e40e6c86c0c079bea612b4eb908f6ec5/pkg/predicate/predicate.go#L56 but looks like that is outdated now and kubernetes has it the way you're suggesting: https://github.com/kubernetes-sigs/controller-runtime/blob/bdb3f845d6e2510d441f593fb420781e8a05f894/pkg/predicate/predicate.go#L170

So I'll go ahead and use this suggestion.